fvm/environment/account_key_reader.go

```diff
@@ -121,35 +134,71 @@ func (reader *accountKeyReader) GetAccountKey(
 			return nil, nil
 		}
 
-		return nil, fmt.Errorf("getting account key failed: %w", err)
+		return formatErr(err)
 	}
 
 	// Prepare the account key to return
+	runtimeAccountKey, err := FlowToRuntimeAccountKey(publicKey)
+	if err != nil {
+		return formatErr(err)
+	}
+
+	return runtimeAccountKey, nil
+}
 
-	signAlgo := crypto.CryptoToRuntimeSigningAlgorithm(publicKey.SignAlgo)
+func (reader *accountKeyReader) AccountKeysCount(address runtime.Address) (uint64, error) {
+	defer reader.tracer.StartSpanFromRoot(trace.FVMEnvAccountKeysCount).End()
+
+	formatErr := func(err error) (uint64, error) {
+		return 0, fmt.Errorf("fetching account key count failed: %w", err)
+	}
+
+	err := reader.meter.MeterComputation(ComputationKindAccountKeysCount, 1)
+	if err != nil {
+		return formatErr(err)
+	}
+
+	accountAddress := flow.Address(address)
+
+	ok, err := reader.accounts.Exists(accountAddress)
+	if err != nil {
+		return formatErr(err)
+	}
+
+	if !ok {
+		return formatErr(errors.NewAccountNotFoundError(accountAddress))
+	}
```

Comment:
This part is probably the responsibility of `accounts.GetPublicKeyCount`, and it does look like it already contains that check in `func (a *StatefulAccounts) getAccountStatus(address flow.Address) (*AccountStatus, error) {` 

Though I see that this is copied logic from `GetAccountKey` where the same argument can be made, so this can be addressed in a separate PR.

Reply:
yeah you caught me cargo-culting code 😓 i can fix both, gimme a sec

Reply:
addressed in #3463